paddle/operators/send_op.cc

```diff
@@ -37,6 +37,7 @@ class SendOp : public framework::OperatorBase {
     auto ins = Inputs("X");
     auto outs = Outputs("Out");
     std::vector<std::string> epmap = Attr<std::vector<std::string>>("epmap");
+    bool do_get = Attr<bool>("DoGet");
     std::vector<std::string> endpoints =
```

Comment:
`endpoints` seems duplicate with `epmap`, can we remove it. And generate the `endpoints` vector during the iteration of `epmap`?

Reply:
Maybe generating a new vector in each `Run` hurts performance, I'll keep it unchanged in this PR and add an todo comment.

Reply:
Thanks for adding the comment! In terms of the performance, I think we should try avoid premature optimization. If its really a performance issue, we can cache the vector. But I doubt if creating a vector every time will have any performance impact.